shared/archive.go

```diff
@@ -0,0 +1,53 @@
+package shared
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func DetectCompression(fname string) ([]string, string, []string, error) {
+	f, err := os.Open(fname)
+	if err != nil {
+		return nil, "", nil, err
+	}
+	defer f.Close()
+
+	return DetectCompressionFile(f)
+}
+
+func DetectCompressionFile(f io.ReadSeeker) ([]string, string, []string, error) {
```

Comment:
@stgraber I also changed this function over in https://github.com/lxc/lxd/pull/6513 so depending on which one lands first there will need to be some rebasing.

Reply:
Rebased